Document choice to use Ansible Project in copyright lines

DesmondSanctity opened the pull request:
##### SUMMARY
This change is to add default copyright text option to the document when unsure of what to use for copyright

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
https://github.com/ansible/ansible/blame/devel/docs/docsite/rst/dev_guide/developing_modules_documenting.rst

##### ADDITIONAL INFORMATION
This PR fixes issue #51143

DesmondSanctity committed:
Document choice to use Ansible Project in copyright lines

This change is to add default copyright text option to the document when unsure of what to use for copyright

sivel commented:
I/we likely need to revise this.  I contacted Red Hat legal a few months back about whether they had updated guidance.  We've not yet started following it, however this link describes the guidance:

https://www.linuxfoundation.org/en/blog/copyright-notices-in-open-source-software-projects/

Which would appear as one of the following:

- Copyright The Ansible Authors.
- Copyright The Ansible Contributors.
- Copyright Contributors to the Ansible project.

These are close enough to what is being recommended in this PR, however the most significant change is we have been advised to no longer include a year at all in the Copyright statement.

The second significant change is that we no longer accept individual names in the Copyright statement, but only indicate that the "contributors" or "authors" maintain the copyright, which indicates:

- the work is copyrighted;
- the contributors of the code licensed it, but retain ownership of their copyrights; and
- it was licensed for distribution as part of the named project.


> By using a common format, the project avoids having to maintain lists of names of the authors or copyright holders, years or ranges of years, and variations on the (c) symbol

sivel reviewed (changes requested):
See https://github.com/ansible/ansible/pull/78164#issuecomment-1170027166

webknjaz commented on docs/docsite/rst/dev_guide/developing_modules_documenting.rst:
```diff
@@ -55,7 +55,7 @@ When adding a second copyright line for a significant feature or rewrite, add th
     # Copyright: (c) 2015, [Original Contributor(s)]
     # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
-
+    # If you are unsure what to use for copyright, use Copyright: (c) <year>, Ansible Project
```
I'm, pretty sure the empty line should be preserved.
```suggestion
    # If you are unsure what to use for copyright, use Copyright: (c) <current_year>, Ansible Project

```

DesmondSanctity replied:
Fixing this now

DesmondSanctity commented:
> See [#78164 (comment)](https://github.com/ansible/ansible/pull/78164#issuecomment-1170027166)

So, will this be a better way of listing the way to use copyright?

```
    # Copyright The Ansible Authors.
    # Copyright The Ansible Contributors.
    # Copyright Contributors to the Ansible project.
    # If you are unsure what to use for copyright, use Copyright The Ansible Project
```

DesmondSanctity committed:
Maintaining empty line removed initially

The commit adds the empty line removed after docs changes for uniformity

Co-authored-by: Sviatoslav Sydorenko <wk.cvs.github@sydorenko.org.ua>

sivel commented on docs/docsite/rst/dev_guide/developing_modules_documenting.rst:
```diff
@@ -55,6 +55,7 @@ When adding a second copyright line for a significant feature or rewrite, add th
     # Copyright: (c) 2017, [New Contributor(s)]
     # Copyright: (c) 2015, [Original Contributor(s)]
     # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+    # If you are unsure what to use for copyright, use Copyright: (c) <current_year>, Ansible Project
 
```
I cannot make a suggestion to other places higher in the doc, so I'll try to just indicate them here as one big diff.

<details>
<summary>Patch...</summary>
<p>

```diff
diff --git a/docs/docsite/rst/dev_guide/developing_modules_documenting.rst b/docs/docsite/rst/dev_guide/developing_modules_documenting.rst
index c8f37e4a8d..92d3ca3708 100644
--- a/docs/docsite/rst/dev_guide/developing_modules_documenting.rst
+++ b/docs/docsite/rst/dev_guide/developing_modules_documenting.rst
@@ -33,28 +33,25 @@ Begin your Ansible module with ``#!/usr/bin/python`` - this "shebang" allows ``a
 Copyright and license
 =====================
 
-After the shebang and UTF-8 coding, add a `copyright line <https://www.gnu.org/licenses/gpl-howto.en.html>`_ with the original copyright holder and a license declaration. The license declaration should be ONLY one line, not the full GPL prefix.:
+After the shebang and UTF-8 coding, add a `copyright line <https://www.linuxfoundation.org/blog/copyright-notices-in-open-source-software-projects/>`_ with the original copyright holder and a license declaration. The license declaration should be ONLY one line, not the full GPL prefix.:
 
 .. code-block:: python
 
     #!/usr/bin/python
     # -*- coding: utf-8 -*-
 
-    # Copyright: (c) 2018, Terry Jones <terry.jones@example.org>
+    # Copyright: Contributors to the Ansible project
     # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-Major additions to the module (for instance, rewrites) may add additional copyright lines. Any legal review will include the source control history, so an exhaustive copyright header is not necessary.
-Please do not edit the existing copyright year. This simplifies project administration and is unlikely to cause any interesting legal issues.
-When adding a second copyright line for a significant feature or rewrite, add the newer line above the older one:
+Additions to the module (for instance, rewrites) are not permitted to add additional copyright lines other than the default copyright statement if missing:
 
 .. code-block:: python
 
-    #!/usr/bin/python
-    # -*- coding: utf-8 -*-
+   # Copyright: Contributors to the Ansible project
 
-    # Copyright: (c) 2017, [New Contributor(s)]
-    # Copyright: (c) 2015, [Original Contributor(s)]
-    # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+Any legal review will include the source control history, so an exhaustive copyright header is not necessary.
+
+Please do not include a copyright year. If the existing copyright statement includes a year, do not edit the existing copyright year. Any existing copyright header should not be modified without permission from the copyright author.
 
 .. _ansible_metadata_block:
 
```

</p>
</details>

This change brings the documentation into alignment with the guidance from Red Hat legal.  We no longer allow individuals to be added to the Copyright list, and we no longer include the year.  However, we do not modify existing copyright statements.  If a generic copyright is not found, one can be added as needed.

DesmondSanctity replied:
Changes has been effected as suggested!

DesmondSanctity committed:
updating the copyright style guide section

This change brings copyrights for documentation changes into alignment with the guidance from Red Hat legal

samccann commented:
Thanks @DesmondSanctity for the Ansible docs fix!
